cli/cmd/plugin/unmanaged-cluster/semver/semver.go

```diff
@@ -0,0 +1,317 @@
+//nolint
+// Copyright 2018 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Package semver implements comparison of semantic version strings.
+// It has been modified from go source code to accept a string without a leading `v`
+// and does not include unneeded helper functions like `Max` or `Build`
+//
+// The general form of a semantic version string accepted by this package is
+//
+//	MAJOR[.MINOR[.PATCH[-PRERELEASE][+BUILD]]]
+//
+// where square brackets indicate optional parts of the syntax;
+// MAJOR, MINOR, and PATCH are decimal integers without extra leading zeros;
+// PRERELEASE and BUILD are each a series of non-empty dot-separated identifiers
+// using only alphanumeric characters and hyphens; and
+// all-numeric PRERELEASE identifiers must not have leading zeros.
+//
+// This package follows Semantic Versioning 2.0.0 (see semver.org)
+// with one exception: it recognizes
+// MAJOR and MAJOR.MINOR (with no prerelease or build suffixes)
+// as shorthands for MAJOR.0.0 and MAJOR.MINOR.0.
+
+//nolint // This is go source code copied and litely modified, but not for linting
+package semver
+
+import "sort"
+
+// parsed returns the parsed form of a semantic version string.
```

Comment:
sweet :)

can we make a quick comment on why we copy (forked) this?

(ie need to remove v requirement)

Reply:
That's good. I missed it!